domain/src/Plugin/Condition/Domain.php

```diff
@@ -89,9 +124,19 @@ public function evaluate() {
     if (empty($domains) && !$this->isNegated()) {
       return TRUE;
     }
-    $domain = $this->getContextValue('domain');
+    $domain = $this->domainNegotiator->getActiveDomain();
+    //dpm($domain);
     // NOTE: The block system handles negation for us.
     return (bool) in_array($domain->id(), $domains);
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  public function getCacheContexts() {
+    $contexts = parent::getCacheContexts();
+    $contexts[] = 'domain';
```

Comment:
We use the `url.site` cache context, not a domain context, since url.site covers our needs.